compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java

```diff
@@ -2383,6 +2386,22 @@ private void analyzeArgIsolatedness(BLangInvocation invocationExpr, List<BLangEx
         analyzeRestArgsForRestParam(invocationExpr, restArgs, symbol, expectsIsolation);
     }
 
+    private BTupleType getRepresentativeTupleTypeForRemainingArgs(int paramCount, int reqArgCount,
+                                                                  BArrayType arrayType) {
+        int remReqArgCount = paramCount - reqArgCount;
+        List<BType> memTypes = new ArrayList<>(remReqArgCount);
+        BType eType = arrayType.eType;
+        for (int i = 0; i < remReqArgCount; i++) {
+            memTypes.add(eType);
+        }
+
+        if (arrayType.size > remReqArgCount) {
+            return new BTupleType(null, memTypes, eType, 0);
+        }
+
+        return new BTupleType(memTypes);
+    }
```

Comment:
Why is it necessary to add a rest type?
Can't we do as follows
```
      if (arrayType.size > remReqArgCount) {
          return new BTupleType(new ArrayList<>(Collections.nCopies(arrayType.size, eType)));
      }
      return new BTupleType(new ArrayList<>(Collections.nCopies(remReqArgCount, eType)));
```

Reply:
There's a (slight) type checking difference since we need to retrieve each member type vs this way we just get the single rest type and check against it. This may also result in large tuples.